File: Pace/MPMediaItem+ASMHash.m
//
//  MPMediaItem+ASMHash.m
//  Pace
//
//  Created by Andrew Molloy on 5/22/14.
//  Copyright (c) 2014 Andrew Molloy. All rights reserved.
//

#import "MPMediaItem+ASMHash.h"
#import "NSString+ASMCityHash.h"

@implementation MPMediaItem (ASMHash)

-(NSString*)digestString
{
	NSString* artist = [self valueForProperty:MPMediaItemPropertyArtist];
	NSString* album = [self valueForProperty:MPMediaItemPropertyAlbumTitle];
	NSString* title = [self valueForProperty:MPMediaItemPropertyTitle];
	NSNumber* duration = [self valueForProperty:MPMediaItemPropertyPlaybackDuration];

	return [NSString stringWithFormat:@"%@ - %@: %@ (%@)", artist, album, title, duration];
}

-(NSNumber*)digest
{
	return @([self.digestString cityHash64]);
}

@end

File: Pace/MPMediaItem+ASMHash.swift
//
//  MPMediaItem+ASMHash.swift
//  Pace
//
//  Created by Andrew Molloy on 6/5/14.
//  Copyright (c) 2014 Andrew Molloy. All rights reserved.
//

import Foundation

extension MPMediaItem
{
	func digestString() -> String {
		return "\(self.artist) - \(self.albumTitle): \(self.title) (\(self.playbackDuration))"
	}

	func digest() -> NSNumber {
		let digestString : NSString = self.digestString()
		return NSNumber(unsignedLongLong: digestString.cityHash64())
	}
}
